### PR TITLE
[Tizen/Api] basic implementation for single shot

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add base code for single shot model.
Update .spec and add simple testcases.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: tizen-api/src/nnstreamer-single.c [hunk 1]
```diff
@@ -23,39 +23,373 @@
...
+
+  if (!model) {
+    dloge ("The given param, model is invalid.");
+    return NNS_ERROR_INVALID_PARAMETER;
+  }
+
+  model_h = (ml_simpleshot_model *) model;
+
+  if (model_h->src)
+    gst_object_unref (model_h->src);
```
Comment (myungjoo) [inline tizen-api/src/nnstreamer-single.c]: Wouldn't it be safer with ```model_h->src = NULL;```  after unref?

Comment (jaeyun-jung) [reply tizen-api/src/nnstreamer-single.c]: add code to set null after unref the element.

File: tizen-api/src/nnstreamer-single.c [hunk 2]
```diff
@@ -23,39 +23,373 @@
...
+
+  if (model_h->sink)
+    gst_object_unref (model_h->sink);
+
+  gst_tensors_info_free (&model_h->in_info);
+  gst_tensors_info_free (&model_h->out_info);
+
+  return nns_pipeline_destroy (model_h->pipe);
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply tizen-api/src/nnstreamer-single.c]: add code to free pipe and model handle.

File: tizen-api/src/nnstreamer-single.c [hunk 3]
```diff
@@ -23,39 +23,373 @@
...
+ml_model_close (ml_simpleshot_model_h model)
+{
+  ml_simpleshot_model *model_h;
+
+  if (!model) {
+    dloge ("The given param, model is invalid.");
+    return NNS_ERROR_INVALID_PARAMETER;
+  }
+
+  model_h = (ml_simpleshot_model *) model;
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply tizen-api/src/nnstreamer-single.c]: It is necessary (typecast from void ptr)

**Pull request merged**